lib/data/external/GCP/GcpApis/copyObject.js

```diff
@@ -0,0 +1,107 @@
+const async = require('async');
+const request = require('request');
+const { errors } = require('arsenal');
+
+const { logger, JsonError, checkSourceString } = require('../GcpUtils');
+const { logHelper } = require('../../utils');
+
+/**
+ * copyObject - minimum required functionality to perform object copy
+ * for GCP Backend
+ * @param {object} params - update metadata params
+ * @param {string} params.Bucket - bucket name
+ * @param {string} params.Key - object key
+ * @param {string} param.CopySource - source object
+ * @param {function} callback - callback function to call with the copy object
+ * result
+ * @return {undefined}
+ */
+function copyObject(params, callback) {
+    const { CopySource } = params;
+    if (!checkSourceString(CopySource)) {
```

Comment:
You may check first if `CopySource` is `undefined`, then make `checkSourceString` also check if the type is a `string` in order to call `.match` on it. This way you would also catch errors if `CopySource` type is neither `string` nor `undefined`.